r_code/synthetic_dgp.R

```diff
@@ -429,16 +419,16 @@ NoisifyDraw <- function(data_inp, seed, log_output = T, sig_y = 0.2) {
 
 #' Creates the requisite number of factors beyond trend, given frequency.
 #'
-#' @param date_tib Grid of date time, spanning the desired number of periods.
-#' @param freq_inp String indicating time unit; "daily", "weekly", "monthly".
+#' @param date_tib Tibble with columns for the date time, as well as for each
+#'    component of the date (e.g. day) spanning the desired number of periods.
 #' @param shock_name String indicating the type of shock, one of "quarter_num",
 #'    "month_num", "week_num", "day_num".
```

Comment:
It might be good to ensure the shock_name is one of the specified if you are serious about the restriction.  I think the code will work for any value, not just the listed ones.